pkg/provider/azure_vmss.go

```diff
@@ -1628,3 +1636,32 @@ func (ss *scaleSet) EnsureBackendPoolDeleted(service *v1.Service, backendPoolID,
 	isOperationSucceeded = true
 	return nil
 }
+
+// GetNodeCIDRMaskByProviderID returns the node CIDR subnet mask by provider ID.
+func (ss *ScaleSet) GetNodeCIDRMasksByProviderID(providerID string) (int, int, error) {
+	_, vmssName, err := getVmssAndResourceGroupNameByVMProviderID(providerID)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	vmss, err := ss.getVMSS(vmssName, azcache.CacheReadTypeDefault)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	var ipv4Mask, ipv6Mask int
+	if v4, ok := vmss.Tags[VMSetCIDRIPV4TagKey]; ok && v4 != nil {
```

Comment:
what would happen when VMSetCIDRIPV4TagKey is not a valid mask size? e.g. negative value or too big?

Reply:
negative value or value smaller than cluster cidr is checked in `ca. updateNodeSubnetMaskSizes `
too big value is checked in `CidrSet. UpdateSubnetMaskSize `